webdriver/tests/execute_script/promise.py

```diff
@@ -0,0 +1,80 @@
+import pytest
+
+from tests.support.asserts import assert_dialog_handled, assert_error, assert_success
+
+
+def execute_script(session, script, args=None):
+    if args is None:
+        args = []
+    body = {"script": script, "args": args}
+
+    return session.transport.send(
+        "POST", "/session/{session_id}/execute/sync".format(
+            session_id=session.session_id),
+        body)
+
+
+def test_promise_resolve(session):
+    response = execute_script(session, """
+        return Promise.resolve('foobar');
+        """)
+    assert_success(response, "foobar")
+
+
+def test_promise_resolve_delayed(session):
+    response = execute_script(session, """
+        return new Promise(
+            (resolve) => setTimeout(
+                () => resolve('foobar'),
+                50
+            )
+        );
+        """)
+    assert_success(response, "foobar") 
+
+
+def test_promise_reject(session):
+    response = execute_script(session, """
+        return Promise.reject(new Error('my error'));
+        """)
+    assert_error(response, "unknown error")
+    assert response.body["value"]["message"] == "Error: my error"
+
+
+def test_promise_reject_delayed(session):
+    response = execute_script(session, """
+        return new Promise(
+            (resolve, reject) => setTimeout(
+                () => reject(new Error('my error')),
+                50
+            )
+        );
+        """)
+    assert_error(response, "unknown error")
+    assert response.body["value"]["message"] == "Error: my error" 
+
+
+def test_promise_resolve_timeout(session):
+    session.timeouts.script = .1
+    response = execute_script(session, """
+        return new Promise(
+            (resolve) => setTimeout(
+                () => resolve('success'),
+                1000
+            )
+        );
+        """)
+    assert_error(response, "timeout error")
+
+
+def test_promise_reject_timeout(session):
+    session.timeouts.script = .1
+    response = execute_script(session, """
+        return new Promise(
+            (resolve, reject) => setTimeout(
+                () => reject(new Error('my error')),
```

Comment:
Why an `Error`  type here? Is it significant? I’d prefer if we called `reject` without any arguments.

Reply:
This is what I wanted to address with https://github.com/web-platform-tests/wpt/pull/13781#issuecomment-434323660. `Promise.reject` expects a reason as to why the promise was rejected. It doesn't and probably can't specify the type of this reason. For debugging purposes [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject#Description) recommends to use an error instance. I can't find anything specific in the EcmaScript [spec](https://www.ecma-international.org/ecma-262/6.0/#sec-promise.reject) though.

Reply:
OK.